x-pack/test/detection_engine_api_integration/security_and_spaces/rule_execution_logic/query.ts

```diff
@@ -1434,6 +1437,616 @@ export default ({ getService }: FtrProviderContext) => {
+            missingAllFieldsDoc,
+            missingAllFieldsDoc,
+            missingAllFieldsDoc,
+          ]);
+
+          const rule: QueryRuleCreateProps = {
+            ...getRuleForSignalTesting(['ecs_compliant']),
+            query: `id:${id}`,
+            alert_suppression: {
+              group_by: ['agent.name', 'agent.version'],
+              missing_fields_strategy: AlertSuppressionMissingFieldsStrategy.DoNotSuppress,
+            },
+            from: 'now-1h',
+            interval: '1h',
+          };
+
+          const { previewId } = await previewRule({
+            supertest,
+            rule,
+            timeframeEnd: new Date('2020-10-28T06:30:00.000Z'),
+          });
+          const previewAlerts = await getPreviewAlerts({
+            es,
+            previewId,
+            size: 10,
+            sort: ['agent.name', 'agent.version'],
+          });
+          // total 6 alerts = 6 unsuppressed (from docs with at least one missing field)
+          expect(previewAlerts.length).to.eql(6);
+
+          // all alerts are not suppressed and do not have suppress properties
+          previewAlerts.forEach((previewAlert) => {
+            const source = previewAlert._source;
+            expect(source).to.have.property('id', id);
+            expect(source).not.to.have.property(ALERT_SUPPRESSION_DOCS_COUNT);
+            expect(source).not.to.have.property(ALERT_SUPPRESSION_END);
+            expect(source).not.to.have.property(ALERT_SUPPRESSION_TERMS);
+            expect(source).not.to.have.property(ALERT_SUPPRESSION_DOCS_COUNT);
+          });
+        });
+
+        // following 2 tests created to show the difference in 2 modes, using the same data and using suppression time window
+        // rule will be executing 2 times and will create alert during both executions, that will be suppressed according to time window
+        describe('with a suppression time window', async () => {
+          let id: string;
+          const timestamp = '2020-10-28T06:00:00.000Z';
+          const laterTimestamp = '2020-10-28T07:00:00.000Z';
+          beforeEach(async () => {
+            id = uuidv4();
+            // we have 3 kind of documents here:
+            // 1. agent.name: 2 docs with agent-1 in both rule runs
+            // 2. agent.name: 2 docs with agent-2 in second rule run only
+            // 3. agent.name: 3 docs with undefined in both rule runs
+            // if suppress on missing = true - we'll get 3 suppressed alerts
+            // if suppress on missing = false - we'll get 2 suppressed alerts and 3 unsuppressed for missing agent.name
+            const firstDoc = { id, '@timestamp': timestamp, agent: { name: 'agent-1' } };
+            const secondDoc = { id, '@timestamp': laterTimestamp, agent: { name: 'agent-1' } };
+            const thirdDoc = { id, '@timestamp': laterTimestamp, agent: { name: 'agent-2' } };
+            const missingFieldDoc1 = { id, '@timestamp': laterTimestamp };
```

Comment:
Is one of these `missingFieldDocs` intended to have a different timestamp?

Reply:
addressed in this PR https://github.com/elastic/kibana/pull/155839